src/rpc/blockchain.cpp

```diff
@@ -363,6 +364,7 @@ void entryToJSON(UniValue &info, const CTxMemPoolEntry &e)
     info.push_back(Pair("ancestorcount", e.GetCountWithAncestors()));
     info.push_back(Pair("ancestorsize", e.GetSizeWithAncestors()));
     info.push_back(Pair("ancestorfees", e.GetModFeesWithAncestors()));
+    info.push_back(Pair("wtxid", mempool.vTxHashes[e.vTxHashesIdx].first.ToString()));
```

Comment:
Is it guaranteed that the entry exists in `mempool.vTxHashes`? if not, this will potentially cause a leak by adding an empty record.

Reply:
Yes, it's guaranteed -- this is used in compact block reconstruction.

Reply:
Thanks for the explanation, that makes sense.
Maybe we could add an assertion here `assert(mempool.vTxHashes.count(e.vTxHashesIdx))`, just in case. But I don't insist.

Reply:
@laanwj `vTxHashes` is a vector - calling [] out of bounds doesn't add an empty record, it's just undefined behaviour.

Reply:
Oh, oops, never mind.